src/hooks/useSettingsReducer.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useReducer, useState } from 'react';
+import { defaultSettings } from '../node/constants';
+import { encode } from '../node/encoding';
+import { get } from '../node/request';
+import { retrieveSettings, setSettings } from '../node/settings';
+import type { Settings } from '../node/types';
+import { Pin } from '../types';
+
+type Action =
+  | {
+      type: 'SET_SETTINGS';
+      payload: Settings;
+    }
+  | {
+      type: 'SET_SETTINGS_API_KEY';
+      payload: Settings['apiKey'];
+    };
+
+function settingsReducer(settings: Settings, action: Action) {
+  switch (action.type) {
+    case 'SET_SETTINGS': {
+      return action.payload;
+    }
+    case 'SET_SETTINGS_API_KEY': {
+      return {
+        ...settings,
+        apiKey: action.payload,
+      };
+    }
+    default:
+      return settings;
+  }
+}
+
+export const useSettingsReducer = () => {
+  const [settings, dispatch] = useReducer(settingsReducer, defaultSettings);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchSettings = async () => {
+      try {
+        const initialSettings = await retrieveSettings();
```

Comment:
why can the initial settings go in the reducer initialization? that was the point imo?

Reply:
as agreed, needed in useEffect